celery/backends/s3.py

```diff
@@ -48,9 +50,13 @@ def __init__(self, **kwargs):
 
         self.base_path = conf.get('s3_base_path', None)
 
+        self.partition_by_date = conf.get('s3_partition_by_date', False)
+
         self._s3_resource = self._connect_to_s3()
 
     def _get_s3_object(self, key):
+        if self.partition_by_date:
+            key = "date=" + str(datetime.utcnow().date()) + "/" + key
```

Comment:
This should probably use the `date_done` attribute of the task metadata so the key can be looked up.